src/credential-status-manager-base.ts

```diff
@@ -427,6 +443,17 @@ export abstract class BaseCredentialStatusManager {
     return statusCredential;
   }
 
+  // updates status of credential in thread-safe manner
+  async updateStatus({
+    credentialId,
+    credentialStatus
+  }: UpdateStatusOptions): Promise<VerifiableCredential> {
+    const release = await this.lock.acquire();
+    const result = await this.updateStatusUnsafe({ credentialId, credentialStatus });
```

Comment:
I think doing a `try {} finally { release() }` structure might make sense here. (Otherwise, if an error happens in `updateStatusUnsafe()`, release might not get called.

Reply:
Just, seeing this, but good point! I have included this update in the PR.